Mytinerary/View Controllers/InputItineraryViewController.m

```diff
@@ -0,0 +1,88 @@
+//
+//  InputItineraryViewController.m
+//  Mytinerary
+//
+//  Created by ehhong on 7/17/19.
+//  Copyright © 2019 michaelvargas. All rights reserved.
+//
+
+#import "InputItineraryViewController.h"
+#import "Itinerary.h"
+
+@interface InputItineraryViewController ()
+
+@property (weak, nonatomic) IBOutlet UITextField *titleTextField;
+@property (weak, nonatomic) IBOutlet UIDatePicker *startTimeDatePicker;
+@property (weak, nonatomic) IBOutlet UIDatePicker *endTimeDatePicker;
+@property (weak, nonatomic) IBOutlet UITextField *budgetTextField;
+
+@property (strong, nonatomic) UIAlertController *alert;
+
+@end
+
+@implementation InputItineraryViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    
+    // set up alert controller
+    self.alert = [UIAlertController alertControllerWithTitle:@"Error"
+                                                     message:@"This is an alert."
+                                              preferredStyle:UIAlertControllerStyleAlert];
+    
+    UIAlertAction* defaultAction = [UIAlertAction actionWithTitle:@"OK" style:UIAlertActionStyleDefault
+                                                          handler:^(UIAlertAction * action) {}];
+    
+    [self.alert addAction:defaultAction];
+}
+
+- (IBAction)onTapCloseButton:(id)sender {
+    [self dismissViewControllerAnimated:YES completion:nil];
+}
+
+- (IBAction)onTapCreateItinerary:(id)sender {
+    // ensure title is inputted
+    NSString *title = self.titleTextField.text;
+    if ([title isEqualToString:@""]) {
+        self.alert.message = @"Missing itinerary title";
+        [self presentViewController:self.alert animated:YES completion:nil];
+    }
+    
+    // ensure start and end times are valid
+    NSDate *startTime = self.startTimeDatePicker.date;
+    NSDate *endTime = self.endTimeDatePicker.date;
+    if ([startTime compare:endTime] != NSOrderedAscending) {
+        self.alert.message = @"End time must be after start time";
+        [self presentViewController:self.alert animated:YES completion:nil];
+    }
+    
+    // check budget validity
+    NSNumber *budget = @([self.budgetTextField.text floatValue]);
+    if (budget < 0) {
+        self.alert.message = @"Budget cannot be negative";
+        [self presentViewController:self.alert animated:YES completion:nil];
+    }
```

Comment:
might be nice to move all of this validation code into a utility class (ItineraryUtils.m?) so that this validation code is not scattered throughout the app (that way too you can centralize you error messages, could use throws to indicate an error and catch it to send to the alert)

Reply:
Would you mind providing some more details on how to do that? Would it be a plain class that inherits NSObject? Also, since the alert is currently a property specific to this view controller, how would I instantiate an alert that is reusable across view controllers?